src/player.js

```diff
@@ -1902,35 +1908,25 @@ export default class Player extends FakeEventTarget {
     }
   }
 
-  /**
-   * Handles auto play.
-   * @returns {void}
-   * @private
-   */
-  _handleAutoPlay(): void {
-    if (this.isAudio() || this._config.playback.autoplay === false) {
-      this._posterManager.show();
-    }
-    if (this._config.playback.autoplay === true) {
-      const allowMutedAutoPlay = this._config.playback.allowMutedAutoPlay;
-      Player.getCapabilities(this.engineType).then(capabilities => {
-        if (capabilities.autoplay) {
-          onAutoPlay();
-        } else {
-          if (capabilities.mutedAutoPlay) {
-            if (this.muted && !this._fallbackToMutedAutoPlay) {
-              onMutedAutoPlay();
-            } else if (allowMutedAutoPlay) {
-              onFallbackToMutedAutoPlay();
-            } else {
-              onAutoPlayFailed();
-            }
+  autoPlay(): void {
```

Comment:
it is not correct to call this autoPlay, as it is more correct to say this is a non user gesture play AKA programmatic play.
WDYT to put this in the play API and pass to it an object {programmatic: boolean} and if it is yes then call `_handleAutoPlay` and return.
BTW, the check for audio should still show the poster in this case.

Reply:
@RoyBregman ?

Reply:
done

Reply:
please see https://github.com/kaltura/playkit-js/pull/521#discussion_r551856918 - this comment is still relevant and not handled.

> BTW, the check for audio should still show the poster in this case.

was this checked? I guess it should be OK as on initial load if it is audio we will get here and show it - please make sure this is covered and make sure it is a tested use case in QA plan

Reply:
I'm not sure I understand the comment - audio case logic is remained the same. So is autoplay true and false.
The only change in logic here is that I show poster also for autoplay "inview".
This is done because someone can define threshold as 90% and scroll slowly until he reaches that point. We don't want to see a black player in the mean while.